tests/jsmarionette/runner/marionette-js-runner/host/session.js

```diff
@@ -60,14 +61,16 @@ Session.create = function(host, profile, options) {
 
   var startSession = resolveBinary(options)
   .then(function(binary) {
-    return host.request('/start_runner', { binary: binary, options: options });
-  })
-  .then(function(result) {
-    var session = new Session(host, result.id, options);
-    host.sessions[result.id] = session;
-    // Session no longer pending.
-    host.pendingSessions.splice(host.pendingSessions.indexOf(startSession), 1);
-    return session;
+    host.request('/start_runner', { binary: binary, options: options })
+      .then(function(result) {
+        console.log('start-runner result =', result);
+        console.log('start-runner result.id =', result.id);
+        var session = new Session(host, result.id, options);
+        host.sessions[result.id] = session;
+        // Session no longer pending.
+        host.pendingSessions.splice(host.pendingSessions.indexOf(startSession), 1);
+        return session;
+  }, function(reject) { console.log(reject) });
```

Comment:
You can do `x.then(...).catch(function(error) { ... })` -- that will catch `resolveBinary` errors in addition to ones from this block